tests/e2e/streams/streams_tests.py

```diff
@@ -408,19 +472,78 @@ def test_bootstrap_server(producer, topics, connection, transformation):
 
     for topic in topics:
         common.check_vertex_exists_with_topic_and_payload(
-            cursor, topic, SIMPLE_MSG)
+            cursor, topic, SIMPLE_MSG
+        )
 
 
 @pytest.mark.parametrize("transformation", TRANSFORMATIONS_TO_CHECK)
 def test_bootstrap_server_empty(producer, topics, connection, transformation):
     assert len(topics) > 0
     cursor = connection.cursor()
     with pytest.raises(mgclient.DatabaseError):
-        common.execute_and_fetch_all(cursor,
-                                     "CREATE STREAM test "
-                                     f"TOPICS {','.join(topics)} "
-                                     f"TRANSFORM {transformation} "
-                                     "BOOTSTRAP_SERVERS ''")
+        common.execute_and_fetch_all(
+            cursor,
+            "CREATE STREAM test "
+            f"TOPICS {','.join(topics)} "
+            f"TRANSFORM {transformation} "
+            "BOOTSTRAP_SERVERS ''",
+        )
+
+
+@pytest.mark.parametrize("transformation", TRANSFORMATIONS_TO_CHECK)
+def test_set_offset(producer, topics, connection, transformation):
+    assert len(topics) > 0
+    cursor = connection.cursor()
+    common.execute_and_fetch_all(
+        cursor,
+        "CREATE STREAM test "
+        f"TOPICS {topics[0]} "
+        f"TRANSFORM {transformation} "
+        "BATCH_SIZE 1",
+    )
+
+    messages = [f"{i} message" for i in range(1, 21)]
+
+    for message in messages:
+        producer.send(topics[0], message.encode()).get(timeout=60)
+
+    def consume(expected_messages):
+        common.start_stream(cursor, "test")
+        if len(expected_messages) == 0:
+            time.sleep(2)
+
+        for message in expected_messages:
+            assert common.check_one_result_row(
+                cursor, f"MATCH (n: MESSAGE {{payload: '{message}'}}) RETURN n"
+            )
```

Comment:
I didn't communicated my idea clear enough. I would do something like this:
```suggestion
        if len(expected_messages) == 0:
            time.sleep(2)
        else:
            assert check_one_result_row(cursor, f"MATCH (n: MESSAGE {{payload: {messages[-1]}}}) RETURN n")
```

And then I would keep the logic (that was removed in https://github.com/memgraph/memgraph/pull/282/commits/3e99fd0f5ca003200bbc1a14b14d38b5833d9665) to query all the vertices and return them.

The main reason is with the current check the extra messages are allowed, because only the existence of expected messages are checked, but not the absence of extra messages. My original idea was about only to improve the wait time (use `check_one_result_row` only for waiting for the last vertex), and keeping the checking logic as is.

Sorry for the confusion.